src/clib/lib/job_queue/torque_driver.cpp

```diff
@@ -442,20 +464,35 @@ static int torque_driver_submit_shell_job(torque_driver_type *driver,
                 /* The qsub command might fail intermittently for acceptable reasons,
 ￼                  retry a couple of times with exponential sleep.  */
                 int return_value = -1;
-                int sleep_time = 2;
-                int max_sleep_time = pow(sleep_time, 5);
-                // This gives a maximum cumulative sleep of 2+4+8+16+32 = 62 seconds
-                while ((return_value != 0) & (sleep_time <= max_sleep_time)) {
+                int retry_interval = 2; /* seconds */
+                int slept_time = 0;
+                while (return_value != 0) {
                     return_value = util_spawn_blocking(
                         driver->qsub_cmd, stringlist_get_size(remote_argv),
                         (const char **)argv, tmp_std_file, tmp_err_file);
                     if (return_value != 0) {
-                        torque_debug(
-                            driver,
-                            "qsub failed for job %s, retrying in %d seconds",
-                            job_name, sleep_time);
-                        sleep(sleep_time);
-                        sleep_time *= 2;
+                        if (slept_time + retry_interval <= driver->timeout) {
+                            torque_debug(driver,
+                                         "qsub failed for job %s, retrying in "
+                                         "%d seconds",
+                                         job_name, retry_interval);
+                            sleep(retry_interval);
+                            slept_time += retry_interval;
+                            retry_interval *= 2;
+                        } else {
+                            torque_debug(
+                                driver,
+                                "qsub failed for job %s, no (more) retries",
```

Comment:
Wouldn't it be useful to inform how many retrials and the total time before timeout?

Reply:
The user should be aware of the timeout when it is explicit in the config file (and this log is only visible for users that puts the driver into debug mode). Predicting the number of allowed retrials is a little bit cumbersome to compute since it is exponential backoff, do we want all that complexity in c++ code? It will be evident when looking at the debug log in a problematic session.

Reply:
I understand. Thanks for explaining!